### PR TITLE
Feature/timetable vm

### PR DESCRIPTION
**Pull request by Petro2806**

### CODE REVIEW

File: ViewModel/TimeTablePageVM.cs [hunk 1]
```diff
@@ -1,10 +1,34 @@
...
 
 namespace ViewModel
 {
-    class TimeTablePageVM
-    {
-    }
+	public class TimeTablePageVM
+	{
+		ITimeTableService TimeTableService;
+		IPersonService PersonService;
```
Comment (Scaler358) [inline ViewModel/TimeTablePageVM.cs]: Unused

File: ViewModel/TimeTablePageVM.cs [hunk 2]
```diff
@@ -1,10 +1,34 @@
...
+			if (Lessons == null)
+			{
+				Lessons = (List<Timetable>)TimeTableService.LoadTimetableAsync(username).Result;
+			}
+			return Lessons;
+		}
+		public String LecturerName(int LecturerID)
+		{
+			Person Lecturer = TimeTableService.SearchLectorById(LecturerID);
+			return Lecturer.Name[0] + ". " + Lecturer.Surname;
```
Comment (Scaler358) [inline ViewModel/TimeTablePageVM.cs]: Better use format

Comment (Petro2806) [reply ViewModel/TimeTablePageVM.cs]: Such that "return String.Format("{0}. {1}", Lecturer.Name[0], Lecturer.Surname)"?

Comment (Petro2806) [reply ViewModel/TimeTablePageVM.cs]: Such that "return String.Format("{0}. {1}", Lecturer.Name[0], Lecturer.Surname)"?

File: DataServices/DataServiceBase.cs [hunk 3]
```diff
@@ -87,11 +87,11 @@ public async Task<List<string>> LoadQuestionsAsync()
...
         }
 
-        public async Task<IList<Timetable>> LoadTimetableAsync(Person person)
+        public async Task<IList<Timetable>> LoadTimetableAsync(string username)
         {
-            return await _dataSource.Timetable
-                .Where(item => item.Group.Equals(person.Student.Group))
-                .ToListAsync();
+			return await _dataSource.Timetable
+				.Where(item => item.GroupID.Equals(this.LoadStudent(this.LoadLogInPersonAsync(username)).GroupID))
```
Comment (RomanLevkovych) [inline DataServices/DataServiceBase.cs]: do we need `this` in this context?

File: DataServices/Services/ITimeTableService.cs [hunk 4]
```diff
@@ -6,6 +6,7 @@ namespace DataServices.Services
 {
     public interface ITimeTableService
     {
-        Task<IList<Timetable>> LoadTimetableAsync(Person person);
-    }
+        Task<IList<Timetable>> LoadTimetableAsync(string username);
+		Person SearchLectorById(int LecturerID);
```
Comment (RomanLevkovych) [inline DataServices/Services/ITimeTableService.cs]: format

File: View/App.xaml.cs [hunk 5]
```diff
@@ -16,7 +16,7 @@ namespace View
 	public partial class App : Application
 	{
         public static string LastSearch { get; set; }
-        public string username { get; set; }
+		public static string username { get; set; } = "RomanLevkovych";
```
Comment (RomanLevkovych) [inline View/App.xaml.cs]: probably, this is bad idea, to hold default username

File: View/Widgets/TimeTablePageContent.xaml.cs [hunk 6]
```diff
@@ -12,17 +12,61 @@
...
+	/// </summary>
+	public partial class TimeTablePageContent : UserControl
+	{
+		TimeTablePageVM Service;
+		List<String> DaysOfWeek;
+		public TimeTablePageContent()
+		{
+			InitializeComponent();
+
+			DaysOfWeek = new List<String>() { "Monday", "Tuesday", "Wednesday", "Thursday", "Friday" };
```
Comment (RomanLevkovych) [inline View/Widgets/TimeTablePageContent.xaml.cs]: enum

File: View/Widgets/TimeTablePageNoteWindow.xaml.cs [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+	{
+		private string textNamePlaceholder = "Title of your note";
+		private NotesPageVM logic;
+		private TimeTablePageTimeTableNote parent;
+		Note note;
+		public TimeTablePageNoteWindow(TimeTablePageTimeTableNote parent, string SubjectName, Note Note = null)
+		{
+			this.logic = new NotesPageVM(new SQLiteDataService(), App.username);
+			this.Title = "Note";
+			this.parent = parent;
```
Comment (RomanLevkovych) [inline View/Widgets/TimeTablePageNoteWindow.xaml.cs]: `this` is redundant in this context

File: View/Widgets/TimeTablePageNoteWindow.xaml.cs [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+
+			this.note = Note;
+			InitializeComponent();
+			textDeadline.DisplayDateStart = DateTime.Now;
+			subjectName.Text = SubjectName;
+			buttonCancel.Click += ButtonCancelClick;
+			if (note == null)
```
Comment (RomanLevkovych) [inline View/Widgets/TimeTablePageNoteWindow.xaml.cs]: try this: https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/nullable-value-types

File: View/Widgets/TimeTablePageNoteWindow.xaml.cs [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
+				subjectName.Text = note.SubjectID;
+				textMaterials.AppendText(note.Materials);
+				buttonAdd.Click += ButtonAddClickModify;
+			}
+		}
+		private void TextTitleGotFocus(object sender, RoutedEventArgs e)
+		{
+			if (textName.Text == textNamePlaceholder)
+			{
+				textName.Text = "";
```
Comment (RomanLevkovych) [inline View/Widgets/TimeTablePageNoteWindow.xaml.cs]: use `string.Empty` instead of ""

File: View/Widgets/TimeTablePageNoteWindow.xaml.cs [hunk 10]
```diff
@@ -0,0 +1,130 @@
...
+			}
+		}
+		private void ButtonAddClick(object sender, RoutedEventArgs e)
+		{
+			if (validateText())
+			{
+				parent.AddNoteFromString(textName.Text, textDeadline.Text, subjectName.Text, new TextRange(textMaterials.Document.ContentStart, textMaterials.Document.ContentEnd).Text);
```
Comment (RomanLevkovych) [inline View/Widgets/TimeTablePageNoteWindow.xaml.cs]: simplify the bunch of parameters

File: View/Widgets/TimeTablePageContent.xaml.cs (hunk 6 above)
Comment (Scaler358) [reply View/Widgets/TimeTablePageContent.xaml.cs]: enum doesnt work in that situation

Comment (Petro2806) [reply View/Widgets/TimeTablePageContent.xaml.cs]: In DB we have days of week as string. I don't know how to do this with enum

**Review by arihilmir — APPROVED** [review]

**Pull request merged**